components/dashboards-web-component/src/utils/DashboardExportor.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import DashboardAPI from './apis/DashboardAPI';
+
+export default class DashboardExportor {
+    /**
+     * Export Dashboard
+     * @param {string} dashboardName name of the dashboard
+     * @param {string} dashboardURL url of the dashboard
+     */
+    static handleDashboardExport(dashboardName, dashboardURL) {
+        DashboardAPI.ExportDashboardByID(dashboardURL)
+            .then((response) => {
+                const url = window.URL.createObjectURL(
+                    new Blob([JSON.stringify(response.data, undefined, 2)]));
+                const link = document.createElement('a');
+                link.href = url;
+                link.setAttribute('download', dashboardName + '.json');
+                document.body.appendChild(link);
+                link.click();
+                document.body.removeChild(link);
+            })
+            .catch((e) => {
+                console.error(e);
```

Comment:
Lets improve the error log. e.g.
```jsx
console.error(`Exporting dashboard ${dashboardName} with URL '${dashboardURL}' failed.`, e)
```

Reply:
Let's also add a TODO saying we should show a proper error message to the user if the exporting failed.

Reply:
Fixed in #5aa031f